src/main.rs

```diff
@@ -82,30 +82,30 @@ fn save(level: &mut CacheLevel, output: Vector, expr: Expr, n: usize, cache: &Ca
 
     if n > MAX_CACHE_LENGTH {
         for dfs_len in n + 2..=MAX_LENGTH {
-            find_binary_expressions_left(level, cache, dfs_len, n, (&output, &expr));
-            find_binary_expressions_right(level, cache, dfs_len, n, (&output, &expr));
+            find_binary_expressions_left(level, cache, dfs_len, n, &expr);
+            find_binary_expressions_right(level, cache, dfs_len, n, &expr);
         }
         if n + 1 <= MAX_LENGTH {
-            find_unary_expression(level, cache, n + 1, (&output, &expr));
+            find_unary_expression(level, cache, n + 1, &expr);
         }
         if n + 2 < MAX_LENGTH && expr.op < Operator::Parens {
-            save(level, output, Expr::parens((&expr).into()), n + 2, cache);
+            save(level, Expr::parens((&expr).into()), n + 2, cache);
         }
         return;
     }
 
-    insert_to_level(level, output, expr);
+    insert_to_level(level, expr);
 }
 
-fn insert_to_level(level: &mut CacheLevel, output: Vector, expr: Expr) {
-    match level.entry(output) {
-        Entry::Occupied(mut e) => {
+fn insert_to_level(level: &mut CacheLevel, expr: Expr) {
```

Comment:
could you update the rayon version too? there's a call in line 485

Reply:
opsy